models/meshmodel/core/v1alpha1/component.go

```diff
@@ -28,24 +28,26 @@ const (
 type ComponentDefinition struct {
 	ID uuid.UUID `json:"-"`
 	TypeMeta
-	DisplayName string                 `json:"displayName" gorm:"displayName"`
-	Format      ComponentFormat        `json:"format" yaml:"format"`
-	Metadata    map[string]interface{} `json:"metadata" yaml:"metadata"`
-	Model       Model                  `json:"model"`
-	Schema      string                 `json:"schema,omitempty" yaml:"schema"`
-	CreatedAt   time.Time              `json:"-"`
-	UpdatedAt   time.Time              `json:"-"`
+	DisplayName     string                 `json:"displayName" gorm:"displayName"`
+	Format          ComponentFormat        `json:"format" yaml:"format"`
+	Metadata        map[string]interface{} `json:"metadata" yaml:"metadata"`
+	IsNotDeployable bool                   `json:"isNotDeployable" yaml:"isNotDeployable" gorm:"isNotDeployable"`
```

Comment:
@MUzairS15 it's not clear to me that this attribute should be a top-level column, but should perhaps, be an attribute within the metadata.